### DISCUSSION TITLE
Updating preset tag versions in renovate.json

### DISCUSSION THREAD

**Discussion by ryancausey** [renovatebot/renovate#19318] [Request Help] [answered]
Does renovate currently have support for updating the versions of shareable config presets that are specified using the tag syntax?

For example, if I have shared preset repo `gitlab.com/my-cool-org/renovate-config` and a renovate.json file that looks like:

```json
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [
        "gitlab>my-cool-org/renovate-config:automerge-dev#v1.0.0"
    ]
}
```

If I add a new tag and a new GitLab release for v1.0.1, is there a way to get renovate to discover this and update renovate.json in the repo?

**Comment by viceice** [comment, marked as answer]
not yet, you need to use regex manager to update. 🤷‍♂️

**Reply by ryancausey** [reply]
are there any known good examples of using this?

**Reply by viceice** [reply]
not sure, highly depends on used platform. maybe you can find something inside discussions

**Reply by viceice** [reply]
maybe inside discussions 🤷‍♂️

**Comment by NINNiT** [comment]
I apologize for posting here in this resolved discussion, but has anyone found something that works?
Is it possible now for renovate to update shared config presets?

**Reply by ryancausey** [reply]
Here's what I'm using for updating [GitLab CI include references](https://docs.gitlab.com/ee/ci/yaml/#includeproject):

```json
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "description": "This adds a regexManager so we can update references to shared GitLab presets based on GitLab releases.",
    "regexManagers": [
        {
            "fileMatch": [
                "^renovate\\.json5?$",
                "^\\.github/renovate\\.json5?$",
                "^\\.gitlab/renovate\\.json5?$",
                "^\\.renovaterc\\.json$",
                "^\\.renovaterc$"
            ],
            "matchStrings": [
                "\"gitlab>your-path/your-renovate-config-project#(?<currentValue>[^\" \\n\\(]+)",
                "\"gitlab>your-path/your-renovate-config-project:.*#(?<currentValue>[^\" \\n\\(]+)",
                "\"gitlab>your-path/your-renovate-config-project/.*#(?<currentValue>[^\" \\n\\(]+)"
            ],
            "datasourceTemplate": "gitlab-releases",
            "depNameTemplate": "your-path/your-renovate-config-project",
            "depTypeTemplate": "renovate-preset"
        }
    ],
    "packageRules": [
        {
            "matchDepTypes": [
                "renovate-preset"
            ],
            "prPriority": 99
        }
    ]
}
```

**Reply by NINNiT** [reply]
Thanks, appreciate that! However, i still think this is something that _should_ be a first-class citizen within renovate...
